app/models/concerns/foreman_azure_rm/vm_extensions/managed_vm.rb

```diff
@@ -193,6 +165,27 @@ def initialize_vm(vm_hash)
       def create_managed_virtual_machine(vm_hash)
         vm_params = initialize_vm(vm_hash)
         vm_params.network_profile = define_network_profile(vm_hash[:network_interface_card_ids])
+        # If image UUID begins with / it is a custom managed image
+        # Otherwise it is a marketplace URN
+        unless vm_hash[:vhd_path].start_with?('/')
```

Comment:
This has been moved here since we do not require image_id in `new_vm` call and only required for `create_vm`.